tmk_core/protocol/arm_atsam/usb/udi_hid_kbd.c

```diff
@@ -738,7 +738,7 @@ bool udi_hid_raw_receive_report(void) {
 static void udi_hid_raw_report_rcvd(udd_ep_status_t status, iram_size_t nb_rcvd, udd_ep_id_t ep) {
     UNUSED(ep);
 
-    if (status == UDD_EP_TRANSFER_OK && nb_rcvd == UDI_HID_RAW_REPORT_SIZE) {
+    if (status == UDD_EP_TRANSFER_OK && nb_rcvd <= UDI_HID_RAW_REPORT_SIZE) {
         UDI_HID_RAW_RECEIVE(udi_hid_raw_report_recv, UDI_HID_RAW_REPORT_SIZE);
```

Comment:
```suggestion
    if (status == UDD_EP_TRANSFER_OK) {
        UDI_HID_RAW_RECEIVE(udi_hid_raw_report_recv, (uint8_t)nb_rcvd);
```
This should theoretically make the length parameter you get in `raw_hid_receive()` correct, as it is not in LUFA (it turns out it is always `RAW_EPSIZE` bytes long, and so is the buffer - I'll look into that).

You might also need to do a memset somewhere, to clear out the buffer from previous packets.

Reply:
@fauxpark I am not sure where the best place zero out the buffer as I have tracked down `udi_hid_raw_report_set` to some externally defined function (binary blob?).

I have also noticed that the raw_hid_send transmits junk data and have zero'd the buffer there.

Reply:
Actually, considering we now have a working length parameter (I hope), maybe we don't really need to clear the buffer...

Reply:
Marking the conservation resolved then. Please feel free to merge to comment on any other changes,